test/test-harden.js

```diff
@@ -0,0 +1,34 @@
+import test from 'tape';
+import SES from '../src/index';
+
+test.skip('hardenerOptions.fringeSet is used', t => {
+  const ws = new WeakSet();
+  let counter = 0;
+  const fringeSet = {
+    add(obj) {
+      counter += 1;
+      return ws.add(obj);
+    },
+    has(obj) {
+      return ws.has(obj);
+    },
+  };
+  t.ok(SES.makeSESRootRealm({ hardenerOptions: { fringeSet } }));
+  t.ok(counter > 0);
+  t.end();
+});
+
+test.skip('hardenerOptions.fringeSet must support add/has', t => {
+  const makeRealm = hardenerOptions =>
+    SES.makeSESRootRealm({ hardenerOptions });
+  // We need to match against the Realm's TypeError, not ours.
+  t.throws(
+    () => makeRealm({ fringeSet: { add: true, has() {} } }),
+    /^TypeError/,
```

Comment:
We've been using the error class itself rather than the name of the error class. That way, we also test that it is an error from the realm we expect.

Reply:
I can't use the class because apparently the error throws before the realm is fully constructed, so when I do:
```javascript
  t.throws(
    () => makeRealm({ fringeSet: { add: true, has() {} } }),
    TypeError,
  );
```

I get:
```
# hardenerOptions.fringeSet must support add/has
not ok 39 should throw
  ---
    operator: throws
    expected: |-
      [Function: TypeError]
    actual: |-
      [Function: TypeError]
    at: Test._951‍.r.t (/Users/michael/src/SES/test/test-harden.js:25:5)
  ...
```